lib/linter/linter.js

```diff
@@ -2166,15 +2259,46 @@ class Linter {
          * That means anytime a fix is successfully applied, there will be another pass.
          * Essentially, guaranteeing a minimum of two passes.
          */
+        const slots = internalSlotsMap.get(this);
+
+        // Remove lint times from the last run.
+        if (stats) {
+            delete slots.times;
+        }
```

Comment:
I think we should reset `slots.fixPasses` as well:

```js
"use strict";

const { ESLint } = require("eslint");
const eslint = new ESLint({
    stats: true,
    fix: true,
    overrideConfigFile: true,
    overrideConfig: {
        rules: {
            "no-extra-parens": 2
        }
    }
});

(async () => {
    await eslint.lintText("((foo))"); // 2 fix passes
    await eslint.lintText("foo;"); // "TypeError: Cannot read properties of undefined (reading 'parse')" - reads `slots.times.passes[2]["parse"]`
})();

```

Reply:
Done